store/src/main/java/org/apache/rocketmq/store/CommitLog.java

```diff
@@ -799,27 +796,18 @@ public long lockTimeMills() {
     }
 
     /**
-     * Spin util acquired the lock.
+     * acquired the lock and return
      */
-    private void lockForPutMessage() {
+    private PutMessageLock lockForPutMessage() {
         if (this.defaultMessageStore.getMessageStoreConfig().isUseReentrantLockWhenPutMessage()) {
```

Comment:
It could be better if we cache the `isUseReentrantLockWhenPutMessage` and invalidate the dynamic change of this variable.